plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -385,11 +389,13 @@ private ListenableFuture<Void> loadPartition(HivePartitionMetadata partition)
         InputFormat<?, ?> inputFormat = getInputFormat(configuration, schema, false);
         FileSystem fs = hdfsEnvironment.getFileSystem(hdfsContext, path);
         boolean s3SelectPushdownEnabled = shouldEnablePushdownForTable(session, table, path.toString(), partition.getPartition());
-
-        // S3 Select pushdown works at the granularity of individual S3 objects,
-        // therefore we must not split files when it is enabled.
+        // S3 Select pushdown works at the granularity of individual S3 objects for compressed files
+        // and finer granularity for uncompressed files using scan range feature.
+        // Scan range is currently enabled for CSV.
+        boolean isCsv = CSV.equals(getDataType(getDeserializerClassName(schema)).orElse(null));
+        boolean s3SelectSplitCheck = !s3SelectPushdownEnabled || (isUncompressed(inputFormat, path) && isCsv);
```

Comment:
Nit: A verb-like name here would be nice for booleans.

Reply:
Sounds good. Any suggestions? My only idea was "isS3SelectSplittable" 🤣

Reply:
Yes, as long as it is a supported case (uncompressed CSV without quoted delimiters).